nav2_bt_navigator/behavior_trees/navigate_to_pose_w_replanning_and_recovery.xml

```diff
@@ -11,26 +11,35 @@
         <RateController hz="1.0">
           <RecoveryNode number_of_retries="1" name="ComputePathToPose">
             <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased" error_code_id="{compute_path_error_code}"/>
-            <ClearEntireCostmap name="ClearGlobalCostmap-Context" service_name="global_costmap/clear_entirely_global_costmap"/>
+            <Sequence>
+              <WouldAPlannerRecoveryHelp error_code="{compute_path_error_code}"/>
+              <ClearEntireCostmap name="ClearGlobalCostmap-Context" service_name="global_costmap/clear_entirely_global_costmap"/>
+            </Sequence>
           </RecoveryNode>
         </RateController>
         <RecoveryNode number_of_retries="1" name="FollowPath">
           <FollowPath path="{path}" controller_id="FollowPath" error_code_id="{follow_path_error_code}"/>
-          <ClearEntireCostmap name="ClearLocalCostmap-Context" service_name="local_costmap/clear_entirely_local_costmap"/>
+          <Sequence>
+            <WouldAControllerRecoveryHelp error_code="{follow_path_error_code}"/>
+            <ClearEntireCostmap name="ClearLocalCostmap-Context" service_name="local_costmap/clear_entirely_local_costmap"/>
+          </Sequence>
         </RecoveryNode>
       </PipelineSequence>
-      <ReactiveFallback name="RecoveryFallback">
-        <GoalUpdated/>
-        <RoundRobin name="RecoveryActions">
-          <Sequence name="ClearingActions">
-            <ClearEntireCostmap name="ClearLocalCostmap-Subtree" service_name="local_costmap/clear_entirely_local_costmap"/>
-            <ClearEntireCostmap name="ClearGlobalCostmap-Subtree" service_name="global_costmap/clear_entirely_global_costmap"/>
-          </Sequence>
-          <Spin spin_dist="1.57"/>
-          <Wait wait_duration="5"/>
-          <BackUp backup_dist="0.30" backup_speed="0.05"/>
-        </RoundRobin>
-      </ReactiveFallback>
+      <Sequence>
+        <WouldAControllerRecoveryHelp error_code="{follow_path_error_code}"/>
```

Comment:
Not also planner? I think it should check if either are failing

Reply:
Actually, and smoother: We should at minimum add it to the tutorial: https://navigation.ros.org/tutorials/docs/adding_smoother.html with a `WouldASmootherRecoveryHelp`

Or, we have a another BT node that is simply "WouldAnyRecoveryHelp" where we check for **all** the error codes from the planner/smoother/controller/etc so that we don't have to have a structure within our BT for the Smoother/Controller/Planner to enter this global tree. That would seem to make most sense.

So that would mean we should probably have 2 more BT nodes: WouldASmootherRecoveryHelp &  WouldAnyRecoveryHelp

Reply:
I think adding a node `WouldASmootherRecoveryHelp` would be great. The current BT doesn't use the smoother server thus there are no Smoother error codes. 

I will open up a separate PR for the documentation and smoother error code changes.

Reply:
That would be fine, though its only 1 more node analog to the 2 in this PR. I've long since "viewed and approved" most files so it isn't any trouble for me to have a couple of new ones here if its not for you. The docs obviously would be another PR :smile: 

Either way, there's still the matter of the header reformatting in the comments above this that need to be reverted so that automatic license validators and find it properly